server/server.ts

```diff
@@ -107,6 +126,93 @@ app.get('/ping', (_req, res) => {
   res.json('pong');
 });
 
+app.post('/discord/auth', async (req, res) => {
+  const decoded = await validateUserToken(req.body?.uid, req.body?.token);
+  if (!decoded) {
+    return res.status(400).json({ error: 'invalid user token' });
+  }
+  const customer = await getCustomerByEmail(decoded.email as string);
+  const isSubscriber = Boolean(
+    customer?.subscriptions?.data?.find((sub) => sub?.status === 'active')
+  );
+  if (!isSubscriber) {
+    return res.status(400).json({ error: 'not subscribed' });
+  }
+  if (!postgres || !discordBot) {
+    return res.sendStatus(500);
+  }
+  try {
+    const user = await discordBot.assignRole(
+      req.body?.username,
+      req.body?.discriminator
+    );
+    if (!user) {
+      return res.status(400).json({
+        error: 'Discord account not found. Please join our Discord server.',
+      });
+    }
+    // Allow only one Discord account with role per email.
+    // Check if this customer already has a discord account with a role. If so remove that role.
+    const result = await postgres?.query(
+      'SELECT * FROM discord WHERE email = $1',
+      [decoded.email]
+    );
+    if (result.rowCount) {
+      await discordBot.assignRole(
+        result.rows[0].username,
+        result.rows[0].discriminator,
+        true
+      );
+    }
+    await upsertObject(
+      postgres,
+      'discord',
+      {
+        username: req.body?.username,
+        discriminator: req.body?.discriminator,
+        email: decoded.email,
+      },
+      { email: decoded.email }
+    );
+  } catch (e) {
+    console.error(e);
+    return res.sendStatus(500);
+  }
+  return res.status(200).json({});
+});
+
+app.delete('/discord/delete', async (req, res) => {
+  const decoded = await validateUserToken(
+    req.body?.uid as string,
+    req.body?.token as string
+  );
+  if (!decoded) {
+    return res.status(400).json({ error: 'invalid user token' });
+  }
+  if (!postgres || !discordBot) {
+    return res.sendStatus(500);
+  }
+  try {
+    const result = await postgres?.query(
```

Comment:
i think you can delete and return the deleted content?

Reply:
I don't think it returns the deleted rows. The rows array is always empty. Are you sure it does? Maybe I'm missing a query method argument?

Reply:
you need to add RETURNING to your query